components/CompanyRegister.tsx

```diff
@@ -1,128 +1,145 @@
+    if (!isRegistered) {
+      try {
+        await writeAsync();
+        registerCompanyDb(data);
+      } catch (error) {
+        console.error('Error:', error);
+        toast.error('Registration Unsuccessful. Please contact support.');
+      }
+    } else {
       toast.error('Wallet address is already registered');
-      return;
-    }
-    try {
-      await writeAsync();
-    } catch (e) {
-      toast.error('Company registration failed. Please contact support');
     }
   };
 
-  const registerCompanyDb = async () => {
-    const companyData: CreateCompanyItem = {
-      companyId: address as string,
-      name: companyName,
-      email: companyEmail,
-      site: companySite,
-    };
-    const options = {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(companyData),
-    };
+  const registerCompanyDb = async (data: FormData) => {
+    const validatedData = schema.safeParse(data);
 
-    try {
-      const response = await fetch('api/companies', options);
-      if (response.ok) {
-        const responseData = await response.json();
+    if (validatedData.success) {
+      const companyData: CreateCompanyItem = {
+        companyId: address as string,
+        name: data.companyName,
+        email: data.companyEmail,
+        site: data.companySite,
+      };
 
-        if (isSuccessResponse(responseData)) {
-          toast.success(`${responseData.item.name} Registered Successfully`);
-          router.push('/');
+      const options = {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(companyData),
+      };
+
+      try {
+        const response = await fetch('api/companies', options);
+        if (response.ok) {
+          const responseData = await response.json();
+          if (isSuccessResponse(responseData)) {
+            toast.success(`${responseData.item.name} Registered Successfully`);
+            router.push('/');
+          } else {
+            toast.error('Registration Unsuccessful. Please contact support.');
+          }
         }
-      } else {
-        toast.error(`Registration Unsuccessful. Please contact support.`);
+      } catch (error) {
+        console.error('Error:', error);
+        toast.error('Registration Unsuccessful. Please contact support.');
       }
-    } catch (error) {
-      console.error('Error:', error);
-      toast.error(`Registration Unsuccessful. Please contact support.`);
+    } else {
+      toast.error('Invalid input. Please check the form.');
     }
   };
 
-  const handleCompanyNameChange = (event: any) => {
-    setCompanyName(event.target.value);
-  };
-
-  const handleCompanyEmailChange = (event: any) => {
-    setCompanyEmail(event.target.value);
-  };
-
-  const handleCompanySiteChange = (event: any) => {
-    setCompanySite(event.target.value);
-  };
-
-  useEffect(() => {
```

Comment:
why has this been removed - it is required

Reply:
this useEffect was here to save to DB once SC had been written (`isSuccess`).

Please stay focussed on changing form validation only!!